cyences_app_for_splunk/bin/sophos_instance_details.py

```diff
@@ -0,0 +1,217 @@
+                response_json = response.json()
+                for i in response_json.get("items"):
+                    SOPHOS_TENANT_DICT[i.get('id')] = i.get('apiHost')
+
+        else:
+            self.logger.error("Error while fetching Tenant Details: {}".format(response.json()))
+            exit()
+
+
+    def get_tenant_list(self,barier_token):
+
+        header = {"Authorization":"Bearer {}".format(barier_token)}
+        response = requests.get(SOPHOS_WHOAMI_URL, headers=header)
+
+        if(response.status_code == 200):
+            response_json = response.json()
+            if(response_json.get("idType")=="tenant"):
+                SOPHOS_TENANT_DICT[response_json.get("id")] = response_json.get("apiHosts").get("dataRegion")
+            elif(response_json.get("idType")=="organization"):
+                self.get_tenant_from_organization(barier_token,response_json.get("id"))
+        else:
+            self.logger.error("Error while fetching Tenant Details: {}".format(response.json()))
+
+
+    def get_instance_uuid(self,barier_token,ip="",hostname="",uuid=""):
+        
+        try:
+            list_of_uuid = []
+
+            if(uuid!="" and uuid!=" "):
+                for tenant in SOPHOS_TENANT_DICT:
+                    requestHeaders = {
+                        "X-Tenant-ID": tenant,
+                        "Authorization": "Bearer "+ barier_token,
+                        "Accept": "application/json"
+                    }
+                    response = requests.get(SOPHOS_TENANT_DICT[tenant]+"/endpoint/v1/endpoints/"+str(uuid), headers=requestHeaders)
+                    response_json = response.json()
+                    if response.status_code == 200:
+                        list_of_uuid.append(response_json)
+                        return list_of_uuid
+                    else:
+                        self.logger.error("Error while fetching UUID of instance")
+
+            elif(hostname!="" and ip!="" and hostname!=" " and ip!=" "):
+                for tenant in SOPHOS_TENANT_DICT:
+                    requestHeaders = {
+                        "X-Tenant-ID": tenant,
+                        "Authorization": "Bearer "+ barier_token,
+                        "Accept": "application/json"
+                    }
+                    response = requests.get(SOPHOS_TENANT_DICT[tenant]+"/endpoint/v1/endpoints?hostnameContains="+hostname+"ipAddresses="+ip, headers=requestHeaders)
+                    response_json = response.json()
+                    if response.status_code == 200:
+                        for instance in response_json.get("items"):
+                            list_of_uuid.append(instance)
+                    else:
+                        self.logger.error("Error while fetching UUID of instance")
+
+            elif(ip!="" and ip!=" "):
+                for tenant in SOPHOS_TENANT_DICT:
+                    requestHeaders = {
```

Comment:
This part of the code seems to be repeating:
```
requestHeaders = {
                        "X-Tenant-ID": tenant,
                        "Authorization": "Bearer "+ barier_token,
                        "Accept": "application/json"
                    }
                    response = requests.get(SOPHOS_TENANT_DICT[tenant]+"/endpoint/v1/endpoints?ipAddresses="+ip, headers=requestHeaders)
                    response_json = response.json()
                    if response.status_code == 200:
                        for instance in response_json.get("items"):
                            list_of_uuid.append(instance)
                    else:
                        self.logger.error("Error while fetching UUID of instance")
```

We can create a function and handle all there. Pass required parameter. And return only `response_json.get("items")`